packages/cli/generation/ir-generator/src/examples/validateUnionExample.ts

```diff
@@ -61,11 +64,24 @@ export function validateUnionExample({
         ];
     }
 
+    const unionBaseProperties = {
+        ...rawUnion["base-properties"]
+    };
+
     const type =
         typeof singleUnionTypeDefinition === "string" ? singleUnionTypeDefinition : singleUnionTypeDefinition.type;
 
     if (typeof type !== "string") {
-        return getRuleViolationForExtraProperties(nonDiscriminantPropertyExamples);
+        return validateUnionBaseProperties({
+            typeName,
+            unionBaseProperties,
+            nonDiscriminantPropertyExamples,
+            typeResolver,
+            exampleResolver,
+            file,
+            workspace,
+            breadcrumbs
+        });
     }
```

Comment:
is this diff right? `getRuleViolationForExtraProperties` -> `validateUnionBaseProperties` ?

Reply:
Yes, but I think we could make this a little more clear. The `validateUnionBaseProperties` function calls `getRuleViolationForExtraProperties` when `unionBaseProperties` is empty which preserves the intended behavior. 

We should hoist that logic out of the function though because it's otherwise a little difficult to follow. I'll update.